src/openrct2/core/Compression.cpp

```diff
@@ -84,7 +85,8 @@ namespace OpenRCT2::Compression
 
     bool zlibDecompress(IStream& source, uint64_t sourceLength, IStream& dest, uint64_t decompressLength, ZlibHeaderType header)
     {
-        Guard::Assert(sourceLength <= source.GetLength() - source.GetPosition());
+        if (sourceLength > source.GetLength() - source.GetPosition())
+            throw IOException("Not Enough Data to Deompress");
```

Comment:
There was a typo here "Deompress" -> "Decompress"

Reply:
oops, that's what I get for rushing through the actual code change, lol. Since it's been merged, it's too late to make the change here, so feel free to make the PR (or just directly commit) the correction yourself.

Reply:
Also missed that, so that's on me.